File: May-18/11-05-18/java/01-Selection-sort.java

import java.util.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        int n, k;
        n = input.nextInt();
        k = input.nextInt();
        
        int ar[] = new int[n];
        for(int i = 0; i < n; i++)
            ar[i] = input.nextInt();
            
        for(int i = 0; i < k; i++)
        {
            int pos = i;
            for(int j = i + 1; j < n; j++)
            {
                if(ar[j] < ar[pos])
                {
                    pos = j;
                }
            }
            int tmp = ar[i];
            ar[i] = ar[pos];
            ar[pos] = tmp;
        }
        for(int i = 0; i < n; i++)
        {
            System.out.print(ar[i] + " ");
        }
    }
}

File: May-18/11-05-18/python/01-Selection-sort.py
n, k = map(int, input().split())

ar = [int(x) for x in input().split()]

for i in range(0, k):
    pos = i
    for j in range(i+1, n):
        if(ar[j] < ar[pos]):
            pos = j
    tmp = ar[i]
    ar[i] = ar[pos]
    ar[pos] = tmp


print(*ar)
    
